clientdb/batch_snapshot.go

```diff
@@ -465,6 +488,34 @@ func deserializeLocalBatchSnapshot(r io.Reader) (*LocalBatchSnapshot, error) {
 		b.MatchedOrders[nonce] = append(b.MatchedOrders[nonce], m)
 	}
 
+	// Older batches had a single clearing price instead of a map. If we
+	// have a non-zero price, it means this is an old snapshot and we don't
+	// need to read any further.
+	if clearingPrice > 0 {
+		b.ClearingPrices[order.LegacyLeaseDurationBucket] = clearingPrice
+
```

Comment:
Hmm, I thought I already replied to this, but apparently that comment got lost. I added the newline on purpose to make the return stand out a bit more since it's an early exit from the function.